src/fund/trading/Trading.sol

```diff
@@ -168,10 +154,34 @@ contract Trading is TokenUser, Spoke, TradingSignatures {
         exchanges.push(Exchange(_exchange, _adapter));
     }
 
+    function __getRiskManagementArgs(
+        uint256 _exchangeIndex,
+        bytes4 _methodSelector,
+        bytes memory _encodedArgs
+    )
+        internal
+        returns (address[6] memory, uint256[3] memory)
+    {
+        if (_methodSelector == TAKE_ORDER) {
+            address adapter = exchanges[_exchangeIndex].adapter;
+            (bool success, bytes memory returnData) = adapter.delegatecall(
+                abi.encodeWithSelector(
+                    OrderTaker(adapter).extractTakeOrderRiskManagementArgs.selector,
+                    _encodedArgs
+                )
+            );
+            require(success, "Encoded arguments might not match");
```

Comment:
see PR-level comment about how we might move this into OrderTaker (or OrderFIller)